src/main/java/de/retest/recheck/cli/subcommands/Commit.java

```diff
@@ -82,6 +84,37 @@ public void run() {
 		}
 	}
 
+	private void checkForWarningAndApplyChanges( final ReviewResult reviewResult ) throws IOException {
+		final List<Boolean> shouldApplyChanges = new ArrayList<>( Arrays.asList( false ) );
+		for ( final AttributeDifference attributeDifference : reviewResult.getAllAttributeDifferences() ) {
+			if ( attributeDifference.getElementIdentificationWarning() != null ) {
+				shouldApplyChanges.add( true );
+			}
+		}
+		if ( shouldApplyChanges.contains( true ) ) {
```

Comment:
In line 89 you iterate over _all_ diffs, but as long there is a single warning, you want to invoke the util. So, you can use a short-circuit operation:

```java
final boolean containsWarnings = reviewResult.getAllAttributeDifferences().stream() //
		.anyMatch( attributeDifference -> attributeDifference.getElementIdentificationWarning() != null );
if ( containsWarnings ) {
	WarningUtil.logWarnings( reviewResult );
	logger.info( "Are you sure you want to continue? [Yes/No] or [Y/N]" );
	inputDataInformation( reviewResult );
} else {
	applyChanges( createSutStatePersistence(), reviewResult );
}
```

BTW: In the future, we probably wanna have something like `AttributeDifference#hasElementIdentificationWarning()` to improve overall handling.